core/src/main/java/com/scalar/db/api/ScanBuilder.java

```diff
@@ -511,13 +785,34 @@ private void checkNotScanWithIndexOrScanAll() {
       }
     }
 
-    private void checkNotScanOrScanAll() {
+    private void checkScanWithIndex() {
       if (!isScanWithIndex) {
         throw new UnsupportedOperationException(
             "This operation is supported only when scanning records of a database using a secondary index.");
       }
     }
 
+    private void checkNotScanWithIndex() {
+      if (isScanWithIndex) {
+        throw new UnsupportedOperationException(
+            "This operation is not supported when scanning records of a database using a secondary index.");
+      }
+    }
+
+    private void checkScanAll() {
+      if (!isScanAll) {
+        throw new UnsupportedOperationException(
+            "This operation is supported only when scanning all the records of a database.");
+      }
+    }
+
+    private void checkConditionsEmpty() {
+      if (!conjunctions.isEmpty()) {
+        throw new UnsupportedOperationException(
+            "This operation is supported only when no conditions are specified at all.");
```

Comment:
Instead of `UnsupportedOperationException`, I think `IllegalStateException` is proper in this case. And I think we can suggest using `clearConditions()` to clear the conditions in the message:
```suggestion
        throw new IllegalStateException(
            "This operation is supported only when no conditions are specified at all. "
                + "Please use clearConditions() to clear all the conditions.");
```

Reply:
Exactly. Fixed in d3bb194.